apis/centoswebpanel_response.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+class CentoswebpanelResponse
+{
+    private $status;
+    private $raw;
+    private $response;
+    private $errors;
+    private $headers;
+
+    /**
+     * CentoswebpanelResponse constructor.
+     *
+     * @param string $apiResponse
+     */
+    public function __construct($apiResponse)
+    {
+        $this->raw = isset($apiResponse['content']) ? $apiResponse['content'] : '';
+        $this->headers = isset($apiResponse['headers']) ? $apiResponse['headers'] : '';
+        $response = json_decode($this->raw);
+        $this->status = 200;
+        $this->response = $response;
+
+        if ($response && $response->status == 'Error') {
+            $this->errors = $response->msj;
+        }
+
+        if (!empty($this->headers)) {
+            $headerOne = explode(' ', $this->headers[0]);
+            if (count($headerOne) >= 2) {
+                $this->status = $headerOne[1];
```

Comment:
The HTTP code is available via cURL, so it could be set by the request rather than parsed from the headers.

```
$httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
```

Are the headers necessary for any purpose other than to get the HTTP code?

Reply:
The other headers are not really necessary I suppose, though the response object does store the headers and makes them accessible to the module